### PR TITLE
Java Mavlink V2

### PR DESCRIPTION
**Pull request by magicrub**
This adds to and closes PRs https://github.com/ArduPilot/pymavlink/pull/320 and https://github.com/ArduPilot/pymavlink/pull/317 which builds from https://github.com/ArduPilot/pymavlink/pull/290 which builds from https://github.com/ArduPilot/pymavlink/pull/127

It's a shame PR #320 didn't allow me to just add this commit on top.

Credited authors:
@HefnySco @Hunter522 @rrr6399 @kellyschrock

I've been using this for a bit and seems to work. Can everyone else jump on this and test so we can get this MAVLink v2 stuff merged and move on. I think it's safe to merge because it only effects MAVLink 2 on java, which is currently not even supported so it won't effect any existing implementations.

### CODE REVIEW

**Comment by magicrub** [comment]
basically, everything is supported except signing. This includes a common input parser (like in C implementation), trimming output, allowing trimmed input, ID >255, and extensions.

**Comment by rrr6399** [comment]
> This adds to and closes PRs #320 and #317 which builds from #290 which builds from #127
> 
> It's a shame PR #320 didn't allow me to just add this commit on top.

@magicrub Sorry, I missed your message on merging your changes. This approach works out fine though.  Thanks!

**Comment by magicrub** [comment]
image



I was actually referring to this, but I'm not a maintainer of this repo so we're just stuck until the pyMavlink Gods can help. For ArduPilot I would have just pushed to your branch and we would have been good to go to test your branch with my change, or my rebase on master would be active on this PR

**Comment by magicrub** [comment]
the only thing I'd want to change is regarding the isMavlink2 IF around the extension population. Most of the classes have an empty if so we should not generate the if if it's empty. Minor thing, it can wait for after this goes in. Best to stop picking at this and focus on getting mainline mavlink 2 support in

**Comment by amilcarlucas** [comment]
Why not integrate #190 as well ?

**Comment by magicrub** [comment]
@amilcarlucas yup, that looks like a good one.

**Comment by magicrub** [comment]
ping

**Comment by amilcarlucas** [comment]
I vote for merging after #190 is integrated and some tests are done.

**Comment by magicrub** [comment]
@amilcarlucas since you insisted, I incorporated yet another PR #190 into this PR and rebased on master. 
LET THE TESTING BEGIN!!!! (some more)

**Comment by magicrub** [comment]
ping! has anyone else tried this yet?

**Comment by hamishwillee** [comment]
@magicrub Just FYI, no I have not, and I am unlikely to do so in a timely manner. Do we know of anyone who is already using this for v1 who could start playing with v2 changes?

**Comment by rrr6399** [comment]
I'll give a try this week.

**Comment by magicrub** [comment]
I haven't tried it since I merged 190 into it. Not even sure if it builds. The more baggage we drag into this the harder it gets to merge.

**Comment by amilcarlucas** [comment]
The CI fails :(

**Comment by magicrub** [comment]
Since merging 190 was your fault, I vote you to fix it :)

File: generator/java/lib/Signature.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/* AUTO-GENERATED FILE.  DO NOT MODIFY.
+ *
+ * This class was automatically generated by the
+ * java mavlink generator tool. It should not be modified by hand.
+ */
+
+package com.MAVLink;
```
Comment (jorgenman) [inline generator/java/lib/Signature.java]:
```suggestion
package com.mavlink;
```

**Comment by magicrub** [comment]
Thanks @jorgenman , I fixed that one thing and rebased aaaaand now CI is failing on something weird. Does anyone understand the error?

**Comment by jorgenman** [comment]
> Thanks @jorgenman , I fixed that one thing and rebased aaaaand now CI is failing on something weird. Does anyone understand the error?

I'm not very familiar with this codebase, but it looks like in `setup.py` there's a reference to `java/lib/Messages/*.*` that I think should change to `java/lib/messages/*.*`?

**Comment by magicrub** [comment]
ahh yes, good eye!

**Comment by magicrub** [comment]
uh oh, GitHub is broken. After I pushed a commit to fix setup.py it seems to has forgotten my branch!
image

**Comment by magicrub** [comment]
well, I'm running this now and don't see any problems. This PR will cause everyone using Java to rename alllllll their includes to use the lowercase form. Other than that hassle, i think this is good to go.

However, GitHub has somehow decided the the branch is not on a different fork. it's looking for it on the ardupilot repo which is not possible. I guess I'll yet-again replace this PR with a new one of the same branch

**Pull request closed without merging**